src/utils/api.js

```diff
@@ -20,14 +39,49 @@ let newListing = {
   condition: "",
   location: "",
   tradeRequired: false,
-  user_id: ""
+  user_id: "",
 };
 
+let listingRef;
 export const postListing = async (event, newListing) => {
   event.preventDefault();
-
   const docRef = await addDoc(collection(db, "listings"), newListing);
   console.log("Document written to Listings with ID: ", docRef.id);
+  listingRef = docRef.id;
+  uploadImage(event);
+  return listingRef;
+};
+
+const storageRef = ref_storage(storage);
+const metadata = {
+  contentType: "image/jpeg",
+};
+
+export const uploadImage = (event) => {
+  event.preventDefault();
+  const file = event.target[6].files[0];
+  if (!file) return;
+  const storageRef = ref_storage(
+    storage,
+    `/files/${uid}/${listingRef}/${file}`
+  );
+  const uploadTask = uploadBytesResumable(storageRef, file, metadata)
+    .then((snapshot) => {
+      return snapshot;
+    })
+    .then((uploadTaskSnapshot) => {
+      getDownloadURL(uploadTaskSnapshot.ref).then((downloadURL) => {
+        updateListingWithImage(downloadURL);
+      });
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+const updateListingWithImage = (downloadURL) => {
```

Comment:
add listing ref as argument here as its passed down from upload image function. This will keep the data within local scope where its needed and tidy.